builtin/credential/approle/path_role.go

```diff
@@ -523,9 +523,9 @@ func (b *backend) pathRoleSecretIDList(req *logical.Request, data *framework.Fie
 	// If the argument to secretIDLock does not start with 2 hex
 	// chars, a generic lock is returned. So, passing empty string
 	// to get the "custom" lock that could be used for listing.
-	lock := b.secretIDLock("")
-	lock.RLock()
-	defer lock.RUnlock()
+	customLock := b.secretIDLock("")
+	customLock.RLock()
```

Comment:
Shouldn't this simply not get locked at this point in time? What is it protecting?